### PR TITLE
Updated density-fitting code

### PR DESCRIPTION
**Pull request by Samragni-93**

### CODE REVIEW

File: pyscf/adc/radc.py [hunk 1]
```diff
@@ -97,42 +100,61 @@ def compute_amplitudes(myadc, eris):
...
-    t1_2 -= 0.5*lib.einsum('kcad,ikcd->ia',eris_ovvv,t2_1_a,optimize=True)
-    t1_2 += lib.einsum('kdac,ikcd->ia',eris_ovvv,t2_1,optimize=True)
-    del eris_ovvv
-    t1_2 -= 0.5*lib.einsum('lcki,klac->ia',eris_ovoo,t2_1_a,optimize=True)
-    t1_2 -= 0.5*lib.einsum('kcli,lkac->ia',eris_ovoo,t2_1_a,optimize=True)
+    for p in range(0,nocc,chnk_size):
+        if getattr(myadc, 'with_df', None):
+            eris_ovvv = dfadc.get_ovvv_df(myadc, eris.Lov, eris.Lvv, p, chnk_size).reshape(-1,nvir,nvir,nvir)
+        else :
+            eris_ovvv = radc_ao2mo.unpack_eri_1(eris.ovvv, nvir)
```
Comment (ssh2) [inline pyscf/adc/radc.py:136]: Does line 136 correspond to in-core algorithm? If so, where is the out-of-core algorithm implementation?

File: pyscf/adc/radc.py [hunk 2]
```diff
@@ -510,6 +655,7 @@ def __init__(self, mf, frozen=0, mo_coeff=None, mo_occ=None):
         self.method = "adc(2)"
         self.method_type = "ip"
         self.with_df = None
+        self.get_mpn_energy = True
```
Comment (ssh2) [inline pyscf/adc/radc.py]: Please rename this to compute_mpn_energy everywhere

File: pyscf/adc/radc.py [hunk 3]
```diff
@@ -573,9 +719,13 @@ def outcore_transform():
 
         eris = self.transform_integrals()
         
-        self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
-        self.e_tot = self.scf_energy + self.e_corr
+        if self.get_mpn_energy == True:
+            self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
+        else:
+            self.t1, self.t2 = compute_amplitudes(self, eris=eris)
+            self.e_corr = 0.0
```
Comment (ssh2) [inline pyscf/adc/radc.py]: Probably should set e_corr to None if it's not computed

File: pyscf/adc/radc.py [hunk 4]
```diff
@@ -616,9 +766,13 @@ def outcore_transform():
 
         eris = self.transform_integrals() 
             
-        self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
-        self.e_tot = self.scf_energy + self.e_corr
+        if self.get_mpn_energy == True:
+            self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
+        else:
+            self.t1, self.t2 = compute_amplitudes(self, eris=eris)
+            self.e_corr = 0.0
```
Comment (ssh2) [inline pyscf/adc/radc.py]: Same here

File: pyscf/adc/radc.py [hunk 5]
```diff
@@ -573,9 +719,13 @@ def outcore_transform():
...
         
-        self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
-        self.e_tot = self.scf_energy + self.e_corr
+        if self.get_mpn_energy == True:
+            self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris=eris, verbose=self.verbose)
+        else:
+            self.t1, self.t2 = compute_amplitudes(self, eris=eris)
+            self.e_corr = 0.0
 
+        self.e_tot = self.scf_energy + self.e_corr
```
Comment (ssh2) [inline pyscf/adc/radc.py]: Do we really need e_tot? Do we ever use it? If not, may be remove it from everywhere.

File: pyscf/adc/radc.py [hunk 6]
```diff
@@ -1288,6 +1723,8 @@ def ea_contract_r_vvvv(myadc,r2,vvvv):
 
 def ea_adc_matvec(adc, M_ab=None, eris=None):
 
+    #print("Starting sigma vector calculation")
```
Comment (ssh2) [inline pyscf/adc/radc.py]: Please remove

File: pyscf/adc/radc.py (hunk 1 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py:136]: Line 136 corresponds to both in-core and out-of-core algorithm. ovvv is packed in the out-of-core algorithm as well

File: pyscf/adc/radc.py (hunk 2 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py]: Done

File: pyscf/adc/radc.py (hunk 3 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py]: Done

File: pyscf/adc/radc.py (hunk 4 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py]: Done

File: pyscf/adc/radc.py (hunk 5 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py]: Removed

File: pyscf/adc/radc.py (hunk 6 above)
Comment (Samragni-93) [reply pyscf/adc/radc.py]: Done

**Pull request merged**